### PR TITLE
Demo test update + Docker file creation for Hop App

### PR DESCRIPTION
**Pull request by sebastienvas**

### CODE REVIEW

File: tests/e2e/framework/kubernetes.go [hunk 1]
```diff
@@ -90,36 +80,41 @@ func newKubeInfo(tmpDir, runID string) *KubeInfo {
...
-		ManagerTag:       *managerTag,
-		CaHub:            *caHub,
-		CaTag:            *caTag,
-		ProxyHub:         *managerHub,
-		ProxyTag:         *managerTag,
+		namespaceCreated: false,
+		MixerImage:       fmt.Sprintf("%s/mixer:%s", *mixerHub, *mixerTag),
+		ManagerImage:     fmt.Sprintf("%s/manager:%s", *managerHub, *managerTag),
+		CaImage:          fmt.Sprintf("%s/ca:%s", *caHub, *caTag),
+		ProxyImage:       fmt.Sprintf("%s/proxy:%s", *managerHub, *managerTag),
```
Comment (yutongz) [inline tests/e2e/framework/kubernetes.go]: Maybe add a comment here explain why we are giving manager hub/tag to proxy, i.e. manager and proxy always share the same.

File: Jenkinsfile [hunk 2]
```diff
@@ -36,23 +36,33 @@ def presubmit(gitUtils, bazel, utils) {
...
+            kubeTestArgs = "-x ${hub},${prSha}"
+            e2eArgs += "--mixer_hub=${hub}  " +
+                "--mixer_tag=${prSha}"
             break
           default:
             break
         }
       }
       sh("tests/kubeTest.sh ${kubeTestArgs}")
+      sh("tests/e2e.sh ${e2eArgs}")
```
Comment (yutongz) [inline Jenkinsfile]: They changed something in manager as well as demo test, such that  our go version of demo test is not compatible with new prs  right now, I am fixing it.

**Comment by yutongz** [comment]
PTAL

File: Jenkinsfile (hunk 2 above)
Comment (sebastienvas) [inline Jenkinsfile]: Will fix in a later PR once I merge from master

File: tests/e2e.sh [hunk 3]
```diff
@@ -26,5 +26,18 @@ function error_exit() {
...
 . ${ROOT}/istio.VERSION || error_exit "Could not source versions"
 
-bazel ${BAZEL_STARTUP_ARGS} run ${BAZEL_RUN_ARGS} //tests/e2e:go_default_test -- ${ARGS} ${@}
-exit $?
+TESTS_TARGETS=($(bazel query 'tests(//tests/e2e/tests/...)'))
+FAILURE_COUNT=0
+SUMMARY='Tests Summary'
+
+for T in ${TESTS_TARGETS[@]}; do
+  bazel ${BAZEL_STARTUP_ARGS} run ${BAZEL_RUN_ARGS} ${T} -- ${ARGS} ${@}
```
Comment (yutongz) [inline tests/e2e.sh]: I think it would be helpful to print a block with the test name, making it clear when we are going to have 20 tests

**Pull request merged**